minigalaxy/installer.py

```diff
@@ -349,12 +374,34 @@ def uninstall_game(game):
         os.remove(path_to_shortcut)
 
 
-def _exe_cmd(cmd):
-    process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    stdout, stderr = process.communicate()
-    stdout = stdout.decode("utf-8")
-    stderr = stderr.decode("utf-8")
-    return stdout, stderr, process.returncode
+def _exe_cmd(cmd, capture_output=True, print_output=False):
+    """Wine commands are very verbose.
+    We should consume the output in regularly to prevent buffers (and minigalaxy memory) from filling up"""
+    print(f'executing wine command: {shlex.join(cmd)}')
+    std_out = []
+    process = subprocess.Popen(cmd,
+                               stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                               bufsize=1, universal_newlines=True, encoding="utf-8")
+    rc = process.poll()
+    while rc is None:
+        out_line = process.stdout.readline()
+        if capture_output and out_line != '':
+            std_out.append(out_line)
+
+        if print_output:
+            print(out_line, end='')
+
+        rc = process.poll()
```

Comment:
What about something like this instead?
```suggestion
    while (rc := process.poll()) is None:
        if data := process.stdout.read():
            stdout.append(data)
            print(data, end='')
        if data := process.stderr.read():
            stdout.append(data)
            print(data, end='')
```

Reply:
Let me do a bit of testing with this.